lib/pos_blockchain.go

```diff
@@ -157,10 +157,13 @@ func (bc *Blockchain) validateAndIndexBlockPoS(block *MsgDeSoBlock) (*BlockNode,
 
 	// Check if the block is properly formed and passes all basic validations.
 	if err = bc.isValidBlockPoS(block); err != nil {
-		blockNode, innerErr := bc.storeValidateFailedBlockInBlockIndex(block)
-		if innerErr != nil {
-			return nil, errors.Wrapf(err, "validateAndIndexBlockPoS: Problem adding validate failed block to block index: %v", innerErr)
+		// It's possible for isValidBlockPoS to return an error that is not a RuleError.
+		// If we have a RuleError, we KNOW that the block should be ValidateFailed.
+		if IsRuleError(err) {
+			return storeValidateFailedBlockInternal(err)
 		}
+		// If we didn't hit a RuleError, then it's unclear whether or not the block should
```

Comment:
Is there a way to clean up `isValidBlockPoS` such that it only returns rule errors? It'll be great to avoid a hazy error case here

Reply:
Will address in a separate PR